src/Uno.UI/UI/Xaml/XamlFilePathHelper.shared.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Extensions;
+
+namespace Uno.UI.Xaml
+{
+	internal static class XamlFilePathHelper
+	{
+		public const string AppXIdentifier = "ms-appx:///";
+		public const string MSResourceIdentifier = "ms-resource:///";
+		public static string LocalResourcePrefix => $"{MSResourceIdentifier}Files/";
+		
+		/// <summary>
+		/// Convert relative source path to absolute path.
+		/// </summary>
+		internal static string ResolveAbsoluteSource(string origin, string relativeTargetPath)
+		{
+			if (IsAbsolutePath(relativeTargetPath))
+			{
+				// The path is already absolute. (Currently we assume it's in the local assembly.)
+				var trimmedPath = relativeTargetPath.TrimStart(AppXIdentifier);
+				return trimmedPath;
+			}
+
+			var originDirectory = Path.GetDirectoryName(origin);
+			if (originDirectory.IsNullOrWhiteSpace())
+			{
+				return relativeTargetPath;
+			}
+
+			var absoluteTargetPath = GetAbsolutePath(originDirectory, relativeTargetPath);
+
+			return absoluteTargetPath.Replace('\\', '/');
+		}
+
+		internal static bool IsAbsolutePath(string relativeTargetPath) => relativeTargetPath.StartsWith(AppXIdentifier)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this call to 'relativeTargetPath.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)